[receiver/SQLServer] Error scraping metrics failed to parse valueKey

### Component(s)

receiver/sqlserver

### What happened?

## Description

While trying to scrape ALL possible metrics using the receiver I am seeing this error:

```
2025-04-02T11:32:19.067-0400    error   scraperhelper@v0.123.0/obs_metrics.go:61        Error scraping metrics  {"error": "failed to parse valueKey for row 330: strconv.ParseInt: parsing \"2.082944e+06\": invalid syntax in Free Space in tempdb (KB)"}
go.opentelemetry.io/collector/scraper/scraperhelper.wrapObsMetrics.func1
        go.opentelemetry.io/collector/scraper/scraperhelper@v0.123.0/obs_metrics.go:61
go.opentelemetry.io/collector/scraper.ScrapeMetricsFunc.ScrapeMetrics
        go.opentelemetry.io/collector/scraper@v0.123.0/metrics.go:24
go.opentelemetry.io/collector/scraper/scraperhelper.scrapeMetrics
        go.opentelemetry.io/collector/scraper/scraperhelper@v0.123.0/controller.go:256
go.opentelemetry.io/collector/scraper/scraperhelper.NewMetricsController.func1
        go.opentelemetry.io/collector/scraper/scraperhelper@v0.123.0/controller.go:228
go.opentelemetry.io/collector/scraper/scraperhelper.(*controller[...]).startScraping.func1
        go.opentelemetry.io/collector/scraper/scraperhelper@v0.123.0/controller.go:171
```

When running the query from https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c10a49fb5853cf63d1bff915e2c9c569f84410e6/receiver/sqlserverreceiver/queries.go#L70 directly and filtering on "Free Space in tempdb (KB)", I see this:

![Image](https://github.com/user-attachments/assets/587b4f59-0704-4302-92d0-fe6474c4aa5b)

## Steps to Reproduce

Run the attached configuration with otel-contrib agent v0.123.0 locally on the target system

### Collector version

v0.123.0

### Environment information

## Environment
- Windows Server 2019
- SQL Server 2019 (RTM-CU29-GDR 15.0.4410.1)


### OpenTelemetry Collector configuration

```yaml
exporters:
  debug:
receivers:  
  sqlserver:
    collection_interval: 60s
    server: ${env:COMPUTERNAME}
    port: 1433    
    username: ${env:SECURE_LOGIN}  
    password: ${env:SECURE_PASSWORD}
    resource_attributes:
      server.address:
        enabled: true
      server.port:
        enabled: true
      sqlserver.computer.name:
        enabled: true
      sqlserver.database.name:
        enabled: true
      sqlserver.instance.name:
        enabled: true      
    metrics:
      sqlserver.batch.request.rate:
        enabled: true
      sqlserver.batch.sql_compilation.rate:
        enabled: true
      sqlserver.batch.sql_recompilation.rate:
        enabled: true
      sqlserver.database.backup_or_restore.rate:
        enabled: true
      sqlserver.database.count:
        enabled: false
      sqlserver.database.execution.errors:
        enabled: true
      sqlserver.database.full_scan.rate:
        enabled: true
      sqlserver.database.io:
        enabled: true
      sqlserver.database.latency:
        enabled: true
      sqlserver.database.operations:
        enabled: true
      sqlserver.database.tempdb.space:
        enabled: true
      sqlserver.database.tempdb.version_store.size:
        enabled: true
      sqlserver.deadlock.rate:
        enabled: true
      sqlserver.index.search.rate:
        enabled: true
      sqlserver.lock.timeout.rate:
        enabled: true
      sqlserver.lock.wait.rate:
        enabled: true
      sqlserver.lock.wait_time.avg:
        enabled: true
      sqlserver.login.rate:
        enabled: true
      sqlserver.logout.rate:
        enabled: true
      sqlserver.memory.grants.pending.count:
        enabled: true
      sqlserver.memory.usage:
        enabled: true
      sqlserver.page.buffer_cache.free_list.stalls.rate:
        enabled: true
      sqlserver.page.buffer_cache.hit_ratio:
        enabled: true
      sqlserver.page.checkpoint.flush.rate:
        enabled: true
      sqlserver.page.lazy_write.rate:
        enabled: true
      sqlserver.page.life_expectancy:
        enabled: true
      sqlserver.page.lookup.rate:
        enabled: true
      sqlserver.page.operation.rate:
        enabled: true
      sqlserver.page.split.rate:
        enabled: true
      sqlserver.processes.blocked:
        enabled: true
      sqlserver.replica.data.rate:
        enabled: true
      sqlserver.resource_pool.disk.throttled.read.rate:
        enabled: true
      sqlserver.resource_pool.disk.throttled.write.rate:
        enabled: true
      sqlserver.table.count:
        enabled: true
      sqlserver.transaction.delay:
        enabled: true
      sqlserver.transaction.mirror_write.rate:
        enabled: true
      sqlserver.transaction.rate:
        enabled: true
      sqlserver.transaction.write.rate:
        enabled: true
      sqlserver.transaction_log.flush.data.rate:
        enabled: true
      sqlserver.transaction_log.flush.rate:
        enabled: true
      sqlserver.transaction_log.flush.wait.rate:
        enabled: true
      sqlserver.transaction_log.growth.count:
        enabled: true
      sqlserver.transaction_log.shrink.count:
        enabled: true
      sqlserver.transaction_log.usage:
        enabled: true
      sqlserver.user.connection.count:
        enabled: true  
    query_sample_collection:
      enabled: false
      max_rows_per_query: 1450
    top_query_collection:
      enabled: false
      lookback_time: 60
      max_query_sample_count: 1000
      top_query_count: 200
service:
  pipelines:
    metrics/sqlserver:
      receivers:
      - sqlserver      
      exporters:
      - debug
  telemetry:
    metrics:
      level: normal
      readers:
      - pull:
          exporter:
            prometheus:
              host: 0.0.0.0
              port: 8888
```

### Log output

```shell
2025-04-02T11:32:17.283-0400    info    service@v0.123.0/service.go:197 Setting up own telemetry...
2025-04-02T11:32:17.283-0400    info    builders/builders.go:26 Development component. May change in the future.
2025-04-02T11:32:17.283-0400    info    service@v0.123.0/service.go:264 Starting otelcol-contrib...     {"Version": "0.123.1", "NumCPU": 8}
2025-04-02T11:32:17.283-0400    info    extensions/extensions.go:41     Starting extensions...
2025-04-02T11:32:17.967-0400    info    service@v0.123.0/service.go:287 Everything is ready. Begin running and processing data.
2025-04-02T11:32:19.067-0400    error   scraperhelper@v0.123.0/obs_metrics.go:61        Error scraping metrics  {"error": "failed to parse valueKey for row 330: strconv.ParseInt: parsing \"2.082944e+06\": invalid syntax in Free Space in tempdb (KB)"}
go.opentelemetry.io/collector/scraper/scraperhelper.wrapObsMetrics.func1
        go.opentelemetry.io/collector/scraper/scraperhelper@v0.123.0/obs_metrics.go:61
go.opentelemetry.io/collector/scraper.ScrapeMetricsFunc.ScrapeMetrics
        go.opentelemetry.io/collector/scraper@v0.123.0/metrics.go:24
go.opentelemetry.io/collector/scraper/scraperhelper.scrapeMetrics
        go.opentelemetry.io/collector/scraper/scraperhelper@v0.123.0/controller.go:256
go.opentelemetry.io/collector/scraper/scraperhelper.NewMetricsController.func1
        go.opentelemetry.io/collector/scraper/scraperhelper@v0.123.0/controller.go:228
go.opentelemetry.io/collector/scraper/scraperhelper.(*controller[...]).startScraping.func1
        go.opentelemetry.io/collector/scraper/scraperhelper@v0.123.0/controller.go:171
```

### Additional context

_No response_

--- a/.chloggen/sqlserver_scientific_notation.yaml
+++ b/.chloggen/sqlserver_scientific_notation.yaml
@@ -1,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: receiver/sqlserver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Properly parse numbers stored in scientific notation to integers
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [39124]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: []

--- a/receiver/sqlserverreceiver/scraper.go
+++ b/receiver/sqlserverreceiver/scraper.go
@@ -160,7 +160,7 @@ func (s *sqlServerScraperHelper) recordDatabaseIOMetrics(ctx context.Context) er
 
 	var errs []error
 	now := pcommon.NewTimestampFromTime(time.Now())
-	var val float64
+	var val any
 	for i, row := range rows {
 		rb := s.mb.NewResourceBuilder()
 		rb.SetSqlserverComputerName(row[computerNameKey])
@@ -169,20 +169,20 @@ func (s *sqlServerScraperHelper) recordDatabaseIOMetrics(ctx context.Context) er
 		rb.SetServerAddress(s.config.Server)
 		rb.SetServerPort(int64(s.config.Port))
 
-		val, err = strconv.ParseFloat(row[readLatencyMsKey], 64)
+		val, err = retrieveFloat(row, readLatencyMsKey)
 		if err != nil {
 			err = fmt.Errorf("row %d: %w", i, err)
 			errs = append(errs, err)
 		} else {
-			s.mb.RecordSqlserverDatabaseLatencyDataPoint(now, val/1e3, row[physicalFilenameKey], row[logicalFilenameKey], row[fileTypeKey], metadata.AttributeDirectionRead)
+			s.mb.RecordSqlserverDatabaseLatencyDataPoint(now, val.(float64)/1e3, row[physicalFilenameKey], row[logicalFilenameKey], row[fileTypeKey], metadata.AttributeDirectionRead)
 		}
 
-		val, err = strconv.ParseFloat(row[writeLatencyMsKey], 64)
+		val, err = retrieveFloat(row, writeLatencyMsKey)
 		if err != nil {
 			err = fmt.Errorf("row %d: %w", i, err)
 			errs = append(errs, err)
 		} else {
-			s.mb.RecordSqlserverDatabaseLatencyDataPoint(now, val/1e3, row[physicalFilenameKey], row[logicalFilenameKey], row[fileTypeKey], metadata.AttributeDirectionWrite)
+			s.mb.RecordSqlserverDatabaseLatencyDataPoint(now, val.(float64)/1e3, row[physicalFilenameKey], row[logicalFilenameKey], row[fileTypeKey], metadata.AttributeDirectionWrite)
 		}
 
 		errs = append(errs, s.mb.RecordSqlserverDatabaseOperationsDataPoint(now, row[readCountKey], row[physicalFilenameKey], row[logicalFilenameKey], row[fileTypeKey], metadata.AttributeDirectionRead))
@@ -253,210 +253,210 @@ func (s *sqlServerScraperHelper) recordDatabasePerfCounterMetrics(ctx context.Co
 
 		switch row[counterKey] {
 		case activeTempTables:
-			val, err := strconv.ParseInt(row[valueKey], 10, 64)
+			val, err := retrieveInt(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, activeTempTables)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverTableCountDataPoint(now, val, metadata.AttributeTableStateActive, metadata.AttributeTableStatusTemporary)
+				s.mb.RecordSqlserverTableCountDataPoint(now, val.(int64), metadata.AttributeTableStateActive, metadata.AttributeTableStatusTemporary)
 			}
 		case backupRestoreThroughputPerSec:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, backupRestoreThroughputPerSec)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverDatabaseBackupOrRestoreRateDataPoint(now, val)
+				s.mb.RecordSqlserverDatabaseBackupOrRestoreRateDataPoint(now, val.(float64))
 			}
 		case batchRequestRate:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, batchRequestRate)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverBatchRequestRateDataPoint(now, val)
+				s.mb.RecordSqlserverBatchRequestRateDataPoint(now, val.(float64))
 			}
 		case bufferCacheHitRatio:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, bufferCacheHitRatio)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverPageBufferCacheHitRatioDataPoint(now, val)
+				s.mb.RecordSqlserverPageBufferCacheHitRatioDataPoint(now, val.(float64))
 			}
 		case bytesReceivedFromReplicaPerSec:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, bytesReceivedFromReplicaPerSec)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverReplicaDataRateDataPoint(now, val, metadata.AttributeReplicaDirectionReceive)
+				s.mb.RecordSqlserverReplicaDataRateDataPoint(now, val.(float64), metadata.AttributeReplicaDirectionReceive)
 			}
 		case bytesSentForReplicaPerSec:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, bytesReceivedFromReplicaPerSec)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverReplicaDataRateDataPoint(now, val, metadata.AttributeReplicaDirectionTransmit)
+				s.mb.RecordSqlserverReplicaDataRateDataPoint(now, val.(float64), metadata.AttributeReplicaDirectionTransmit)
 			}
 		case diskReadIOThrottled:
 			errs = append(errs, s.mb.RecordSqlserverResourcePoolDiskThrottledReadRateDataPoint(now, row[valueKey]))
 		case diskWriteIOThrottled:
 			errs = append(errs, s.mb.RecordSqlserverResourcePoolDiskThrottledWriteRateDataPoint(now, row[valueKey]))
 		case executionErrors:
-			val, err := strconv.ParseInt(row[valueKey], 10, 64)
+			val, err := retrieveInt(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, executionErrors)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverDatabaseExecutionErrorsDataPoint(now, val)
+				s.mb.RecordSqlserverDatabaseExecutionErrorsDataPoint(now, val.(int64))
 			}
 		case freeListStalls:
-			val, err := strconv.ParseInt(row[valueKey], 10, 64)
+			val, err := retrieveInt(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, freeListStalls)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverPageBufferCacheFreeListStallsRateDataPoint(now, val)
+				s.mb.RecordSqlserverPageBufferCacheFreeListStallsRateDataPoint(now, val.(int64))
 			}
 		case fullScansPerSec:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, fullScansPerSec)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverDatabaseFullScanRateDataPoint(now, val)
+				s.mb.RecordSqlserverDatabaseFullScanRateDataPoint(now, val.(float64))
 			}
 		case freeSpaceInTempdb:
-			val, err := strconv.ParseInt(row[valueKey], 10, 64)
+			val, err := retrieveInt(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, freeSpaceInTempdb)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverDatabaseTempdbSpaceDataPoint(now, val, metadata.AttributeTempdbStateFree)
+				s.mb.RecordSqlserverDatabaseTempdbSpaceDataPoint(now, val.(int64), metadata.AttributeTempdbStateFree)
 			}
 		case indexSearchesPerSec:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, indexSearchesPerSec)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverIndexSearchRateDataPoint(now, val)
+				s.mb.RecordSqlserverIndexSearchRateDataPoint(now, val.(float64))
 			}
 		case lockTimeoutsPerSec:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, lockTimeoutsPerSec)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverLockTimeoutRateDataPoint(now, val)
+				s.mb.RecordSqlserverLockTimeoutRateDataPoint(now, val.(float64))
 			}
 		case lockWaits:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, lockWaits)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverLockWaitRateDataPoint(now, val)
+				s.mb.RecordSqlserverLockWaitRateDataPoint(now, val.(float64))
 			}
 		case loginsPerSec:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, loginsPerSec)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverLoginRateDataPoint(now, val)
+				s.mb.RecordSqlserverLoginRateDataPoint(now, val.(float64))
 			}
 		case logoutPerSec:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, logoutPerSec)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverLogoutRateDataPoint(now, val)
+				s.mb.RecordSqlserverLogoutRateDataPoint(now, val.(float64))
 			}
 		case memoryGrantsPending:
-			val, err := strconv.ParseInt(row[valueKey], 10, 64)
+			val, err := retrieveInt(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, memoryGrantsPending)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverMemoryGrantsPendingCountDataPoint(now, val)
+				s.mb.RecordSqlserverMemoryGrantsPendingCountDataPoint(now, val.(int64))
 			}
 		case mirrorWritesTransactionPerSec:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, mirrorWritesTransactionPerSec)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverTransactionMirrorWriteRateDataPoint(now, val)
+				s.mb.RecordSqlserverTransactionMirrorWriteRateDataPoint(now, val.(float64))
 			}
 		case numberOfDeadlocksPerSec:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, numberOfDeadlocksPerSec)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverDeadlockRateDataPoint(now, val)
+				s.mb.RecordSqlserverDeadlockRateDataPoint(now, val.(float64))
 			}
 		case pageLookupsPerSec:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, pageLookupsPerSec)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverPageLookupRateDataPoint(now, val)
+				s.mb.RecordSqlserverPageLookupRateDataPoint(now, val.(float64))
 			}
 		case processesBlocked:
 			errs = append(errs, s.mb.RecordSqlserverProcessesBlockedDataPoint(now, row[valueKey]))
 		case sqlCompilationRate:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, sqlCompilationRate)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverBatchSQLCompilationRateDataPoint(now, val)
+				s.mb.RecordSqlserverBatchSQLCompilationRateDataPoint(now, val.(float64))
 			}
 		case sqlReCompilationsRate:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, sqlReCompilationsRate)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverBatchSQLRecompilationRateDataPoint(now, val)
+				s.mb.RecordSqlserverBatchSQLRecompilationRateDataPoint(now, val.(float64))
 			}
 		case transactionDelay:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, transactionDelay)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverTransactionDelayDataPoint(now, val)
+				s.mb.RecordSqlserverTransactionDelayDataPoint(now, val.(float64))
 			}
 		case userConnCount:
-			val, err := strconv.ParseInt(row[valueKey], 10, 64)
+			val, err := retrieveInt(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, userConnCount)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverUserConnectionCountDataPoint(now, val)
+				s.mb.RecordSqlserverUserConnectionCountDataPoint(now, val.(int64))
 			}
 		case usedMemory:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, usedMemory)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverMemoryUsageDataPoint(now, val)
+				s.mb.RecordSqlserverMemoryUsageDataPoint(now, val.(float64))
 			}
 		case versionStoreSize:
-			val, err := strconv.ParseFloat(row[valueKey], 64)
+			val, err := retrieveFloat(row, valueKey)
 			if err != nil {
 				err = fmt.Errorf("failed to parse valueKey for row %d: %w in %s", i, err, versionStoreSize)
 				errs = append(errs, err)
 			} else {
-				s.mb.RecordSqlserverDatabaseTempdbVersionStoreSizeDataPoint(now, val)
+				s.mb.RecordSqlserverDatabaseTempdbVersionStoreSizeDataPoint(now, val.(float64))
 			}
 		}
 
@@ -831,11 +831,25 @@ func vanillaRetriever(row sqlquery.StringMap, columnName string) (any, error) {
 
 func retrieveInt(row sqlquery.StringMap, columnName string) (any, error) {
 	var err error
-	result := 0
+	var result int64
 	if row[columnName] != "" {
-		result, err = strconv.Atoi(row[columnName])
+		result, err = strconv.ParseInt(row[columnName], 10, 64)
+		if err != nil {
+			// SQL Server stores large integers in scientific e notation
+			// (eg 123456 is stored as 1.23456e+5)
+			// This value cannot be parsed by strconv.ParseInt, but is successfully
+			// parsed by strconv.ParseFloat. The goal is here to convert to int
+			// even if the stored value is in scientific e notation.
+			var resultFloat float64
+			resultFloat, err = strconv.ParseFloat(row[columnName], 64)
+			if err == nil {
+				result = int64(resultFloat)
+			}
+		}
+	} else {
+		err = fmt.Errorf("no value found for column %s", columnName)
 	}
-	return int64(result), err
+	return result, err
 }
 
 func retrieveIntAndConvert(convert func(int64) any) func(row sqlquery.StringMap, columnName string) (any, error) {
@@ -851,6 +865,8 @@ func retrieveFloat(row sqlquery.StringMap, columnName string) (any, error) {
 	var result float64
 	if row[columnName] != "" {
 		result, err = strconv.ParseFloat(row[columnName], 64)
+	} else {
+		err = fmt.Errorf("no value found for column %s", columnName)
 	}
 	return result, err
 }

--- a/receiver/sqlserverreceiver/testdata/expectedPerfCounters.yaml
+++ b/receiver/sqlserverreceiver/testdata/expectedPerfCounters.yaml
@@ -39,7 +39,7 @@ resourceMetrics:
           - description: Number of execution errors.
             gauge:
               dataPoints:
-                - asInt: 1048
+                - asInt: 18256656
                   startTimeUnixNano: "1000000"
                   timeUnixNano: "2000000"
             name: sqlserver.database.execution.errors
@@ -900,7 +900,7 @@ resourceMetrics:
             sum:
               aggregationTemporality: 2
               dataPoints:
-                - asInt: 61824
+                - asInt: 18256656
                   attributes:
                     - key: tempdb.state
                       value:

--- a/receiver/sqlserverreceiver/testdata/perfCounterQueryData.txt
+++ b/receiver/sqlserverreceiver/testdata/perfCounterQueryData.txt
@@ -28,7 +28,7 @@
       "object":"SQLServer:Access Methods",
       "sql_instance":"8cac97ac9b8f",
       "computer_name":"abcde",
-      "value":"1048"
+      "value":"1.8256656e+07"
    },
    {
       "counter":"Index Searches/sec",
@@ -2708,7 +2708,7 @@
       "object":"SQLServer:Transactions",
       "sql_instance":"8cac97ac9b8f",
       "computer_name":"abcde",
-      "value":"61824"
+      "value":"1.8256656e+07"
    },
    {
       "counter":"Version Store Size (KB)",

